src/plugins/data_table/data_table.js

```diff
@@ -317,142 +451,55 @@ var LocalDataTable = (function() {
-    //   these checkboxes need to be identified by the value they represent
-    // The IDs "value", "gt", "lt" and "eq" are used to determine in what element in the
-    // filter object in the column manager we store the value entered by the user
-    _generateFilteringControls: function(head, col) {
-      var table = this;
-      var filter = col.filter;
-      if (filter.type === "string") {
-        $(head).append(table.filterStringTemplate( { title: col.title } ));
-      } else if (filter.type === "numeric") {
-        $(head).append(table.filterNumericTemplate());
-      } else if ((filter.type === "list") && (filter.options)) {
-        var checkList = '<ul>';
-        for (var i = 0; i < filter.options.length; i++)
-          checkList += table.filterListTemplate( { attr: col.attr, options: filter.options[i] } );
-        checkList += '</ul>';
-        $(head).append(checkList);
+    // This toggles a filter icon between filters set icon and no filters set icon
+    _toggleIcon: function(icon, enabled) {
+      icon.removeClass("filterActive");
+      icon.removeClass("filterInactive");
+      if (enabled) {
+        icon.addClass("filterActive");
+      } else {
+        icon.addClass("filterInactive");
       }
     },
 
-    // Here we bind events to the input controls for a particular column for filtering.
-    // This will update the column manager, and the table, when the user requests it.
-    // @head: The DOM element for the thead th we want to bind events for
-    // @col: The column from columnManager that corresponds to the thead th we're
-    //   binding events for.
-    // @table: the table we're binding events for
-    _bindFilteringEvents: function(head, col) {
+    // Sets up filtering for the dataTable
+    _setupFiltering: function() {
       var table = this;
-      var filter = col.filter;
+      var cg = table.configGenerator();
 
-      // bind focus to click event because of unbinding click from thead th when
-      // installing sort interceptors
-      $('input', head).on("click", function () {
-        this.focus();
-      });
-      // update columnManager filter and ajaxUpdate dataTable when input changed
-      $('input', head).on('change', function () {
-        if (filter.type === "list") {
-          if (this.checked) {
-            filter[this.id] = filter[this.id] || [];
-            filter[this.id].push(this.value);
-          }
-          else {
-            var index = filter[this.id].indexOf(this.value);
-            if (index > -1)
-              filter[this.id].splice(index, 1);
-            if (filter[this.id].length === 0)
-              filter[this.id] = null;
-          }
-        } else if (this.value === "") {
-          filter[this.id] = null;
-        } else {
-          filter[this.id] = this.value;
+      // Close active filter menu if user clicks on document
+      $(document).click(function (event) {
```

Comment:
I understand how document event handlers are messy, but there's really no better way that I'm aware of with dropdown-menus, at least.